HPX compiler match too strict?

## Expected Behavior

Do we really need the compiler to match as precise as we check? Would it be sufficient to just check for the major version? So HPX compiled with gcc 9.0.1 and the application compiled with 9.0.2 or 9.1.1 should be fine? 

## Actual Behavior
```

CMake Error at /usr/lib64/cmake/HPX/HPX_Message.cmake:43 (message):
  ERROR: Compilers do not match.  In order to compile HPX application it is
  recommended to use the same compiler as you did for HPX.
  HPX_CXX_COMPILER=/usr/bin/c++, CMAKE_CXX_COMPILER=/usr/bin/c++.
  HPX_CXX_COMPILER_ID=GNU, CMAKE_CXX_COMPILER_ID=GNU.
  HPX_CXX_COMPILER_VERSION=9.0.1, CMAKE_CXX_COMPILER_VERSION=9.1.1.  To
  disable this message set HPX_IGNORE_COMPILER_COMPATIBILITY to On.
Call Stack (most recent call first):
  /usr/lib64/cmake/HPX/HPXMacros.cmake:30 (hpx_error)
  /usr/lib64/cmake/HPX/HPXConfig.cmake:67 (hpx_check_compiler_compatibility)
  CMakeLists.txt:20 (find_package)
```

--- a/cmake/templates/HPXMacros.cmake.in
+++ b/cmake/templates/HPXMacros.cmake.in
@@ -1,5 +1,6 @@
 # Copyright (c) 2014 Thomas Heller
 # Copyright (c) 2015 Andreas Schäfer
+# Copyright (c) 2019 Patrick Diehl
 #
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
@@ -10,7 +11,7 @@ include(GNUInstallDirs)
 include(HPX_Utils)
 
 function(hpx_check_compiler_compatibility)
-  if(HPX_IGNORE_COMPILER_COMPATIBILITY)
+    if(HPX_IGNORE_COMPILER_COMPATIBILITY OR HPX_WITH_BUILD_BINARY_PACKAGE)
     return()
   endif()
 

